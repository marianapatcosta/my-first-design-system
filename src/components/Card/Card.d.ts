import React, { CSSProperties, ReactNode } from 'react';
export interface CardProps {
    /**
     * element id
     */
    id?: string;
    /**
     * additional css class
     */
    className?: string;
    /**
     * additional style
     */
    style?: CSSProperties;
    /**
     * element id
     */
    children?: string | ReactNode;
}
export declare const Card: React.ForwardRefExoticComponent<CardProps & React.RefAttributes<HTMLDivElement>>;
