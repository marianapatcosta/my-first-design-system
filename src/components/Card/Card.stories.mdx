import { Meta } from '@storybook/addon-docs/blocks';
import { Card } from './Card';

<Meta title='Design System|Card' component={Card} />

# Card
Card component

## Usage
Use the card to group element inside a card-like container

- Default Card:
  <Story name='DefaultCard'>
    <Card>
      <div>This is Card component</div>
    </Card>
  </Story>

## Other important notes
not applicable.
