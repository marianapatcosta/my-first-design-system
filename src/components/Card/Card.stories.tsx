import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { Card, CardProps } from './Card';

export default {
  title: 'Card',
  component: Card
} as Meta;

const Template: Story<CardProps> = (args) => <Card {...args} />;

export const DefaultCard = Template.bind({});
DefaultCard.args = {
  children: <div>This is Card component</div>
};