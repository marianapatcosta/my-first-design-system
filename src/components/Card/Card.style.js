var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
import { baseStyles, colors } from '../../styles';
export var StyledCard = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  ", "\n  box-shadow: 0 0.125rem 0.5rem ", ";\n  border-radius: 0.375rem;\n  padding: 1rem;\n  overflow: hidden;\n  background-color: ", ";\n\n  @media (min-width: 768px) {\n    padding: 2rem;\n  }\n"], ["\n  ", "\n  box-shadow: 0 0.125rem 0.5rem ", ";\n  border-radius: 0.375rem;\n  padding: 1rem;\n  overflow: hidden;\n  background-color: ", ";\n\n  @media (min-width: 768px) {\n    padding: 2rem;\n  }\n"])), baseStyles, colors.shadow, colors.primary);
var templateObject_1;
