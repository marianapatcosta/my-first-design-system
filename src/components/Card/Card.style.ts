import styled from 'styled-components';
import { baseStyles, colors } from '../../styles';

export const StyledCard = styled.div`
  ${baseStyles}
  box-shadow: 0 0.125rem 0.5rem ${colors.shadow};
  border-radius: 0.375rem;
  padding: 1rem;
  overflow: hidden;
  background-color: ${colors.primary};

  @media (min-width: 768px) {
    padding: 2rem;
  }
`;
