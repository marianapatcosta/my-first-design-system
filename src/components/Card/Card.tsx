import React, { forwardRef, ReactNode } from 'react';
import { StyledCard } from './Card.style';

export interface CardProps {
  /**
   * element id
   */
  id?: string;
  /**
   * additional css class
   */
  className?: string;
  /**
   * element id
   */
  children?: ReactNode;
}

export const Card = forwardRef<HTMLDivElement, CardProps>(
  ({ children, ...props }, ref) => {
    return (
      <StyledCard ref={ref} {...props}>
        {children}
      </StyledCard>
    );
  }
);

Card.defaultProps = {
  className: '',
  id: '',
  children: null,
};
