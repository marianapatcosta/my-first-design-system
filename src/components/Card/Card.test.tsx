import React from 'react';
import { render, getByText } from '@testing-library/react';
import { Card } from './Card';

describe('Card', () => {
  test('renders without errors and matches snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Card />);

    expect(firstChild).toMatchSnapshot();
  });

  test('renders children ', () => {
    const { container } = render(
      <Card>
        <div id='card-child'>Card Children</div>
      </Card>
    );

    getByText(container, 'Card Children');
  });
});
