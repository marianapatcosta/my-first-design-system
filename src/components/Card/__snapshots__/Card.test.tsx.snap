// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders without errors and matches snapshot 1`] = `
<div
  class="sc-bdfBwQ hUyrMc"
  id=""
/>
`;
